lib/macho/macho_file.rb

```diff
@@ -233,10 +264,12 @@ def dylib_id=(new_id)
       raise ArgumentError.new("new ID must be a String") unless new_id.is_a?(String)
       return unless dylib?
 
-      dylib_cmd = command(:LC_ID_DYLIB).first
-      old_id = dylib_id
+      old_cmd = command(:LC_ID_DYLIB).first
+      new_cmd = LoadCommand.create(:LC_ID_DYLIB, new_id, old_cmd.timestamp,
+        old_cmd.current_version, old_cmd.compatibility_version)
 
-      set_name_in_dylib(dylib_cmd, old_id, new_id)
+      add_command(new_cmd)
+      delete_command(old_cmd)
```

Comment:
Not because deleting a load command will set off the padding condition, but because we don't want an insertion to fail halfway through. If we let the deletion occur successfully (which it always will) and run into a failure during insertion, the resulting object will be left without that load command.